tests/cxx/tests/DREAM/MeanExcitationEnergy.cpp

```diff
@@ -0,0 +1,198 @@
+    real_t *nions = new real_t[N];
+    real_t ncold = 0;
+    real_t ntot = 0;
+    len_t ionOffset = 0, rOffset = 0;
+    for (len_t iIon = 0; iIon < N_IONS; iIon++) 
+        for (len_t Z0 = 0; Z0 <= Z_IONS[iIon]; Z0++, ionOffset++){ 
+            ni =  ION_DENSITY_REF;
+            ncold += Z0*ni;
+            ntot  += Z_IONS[iIon]*ni;
+            for (len_t ir = 0; ir < g->GetNr(); ir++, rOffset++)
+                nions[rOffset] = ni;
+        }
+        
+    
+    uqh->SetInitialValue(DREAM::OptionConstants::UQTY_ION_SPECIES, nions);
+
+    #define SETVAL(NAME, v) do { \
+            for (len_t i = 0; i < g->GetNr(); i++) \
+                temp[i] = (v); \
+            uqh->SetInitialValue((NAME), temp); \
+        } while (false)
+
+    // Set electron quantities
+    real_t *temp = new real_t[g->GetNr()];
+    SETVAL(DREAM::OptionConstants::UQTY_N_COLD, ncold);
+    SETVAL(DREAM::OptionConstants::UQTY_N_HOT,  ncold*1e-12);
+    SETVAL(DREAM::OptionConstants::UQTY_N_TOT,  ntot);
+    SETVAL(DREAM::OptionConstants::UQTY_T_COLD, T_cold);
+    SETVAL(DREAM::OptionConstants::UQTY_F_HOT, 0);
+
+    for(len_t i=0;i<g->GetNr();i++)
+        temp[i] = 20*(30*i+1);
+    uqh->SetInitialValue(DREAM::OptionConstants::UQTY_E_FIELD,temp);
+
+    delete [] nions;
+    delete [] temp;
+    
+    return uqh;
+}
+
+
+/**
+ * Generate a default ion handler.
+ */
+DREAM::IonHandler *MeanExcitationEnergy::GetIonHandler(
+    DREAM::FVM::Grid *g, DREAM::FVM::UnknownQuantityHandler *uqh, const len_t N_IONS, const len_t *Z_IONS
+) {
+    vector<string> tritiumNames(0);
+    vector<string> names(N_IONS);
+    for (len_t i = 0; i < N_IONS; i++)
+        names[i] = "";//ION_NAMES[i];
+
+    return new DREAM::IonHandler(
+        g->GetRadialGrid(), uqh, Z_IONS, N_IONS, names, tritiumNames
+    );
+}
+
+real_t* MeanExcitationEnergy::GetMeanExcitationEnergies(DREAM::CollisionQuantity::collqty_settings *cq, const len_t N_IONS, const len_t *Z_IONS, 
+    const len_t N_SPECIES_TO_TEST, const len_t *Z_TO_TEST, const len_t *Z0_TO_TEST,
+    const real_t ION_DENSITY_REF, const real_t T_cold, const real_t B0, const len_t nr
+){
+    DREAM::FVM::Grid *grid = this->InitializeFluidGrid(nr,B0);
+    
+    DREAM::FVM::UnknownQuantityHandler *unknowns = GetUnknownHandler(grid,N_IONS, Z_IONS, ION_DENSITY_REF,T_cold);
+    DREAM::IonHandler *ionHandler = GetIonHandler(grid,unknowns, N_IONS, Z_IONS);
+    DREAM::OptionConstants::momentumgrid_type gridtype = DREAM::OptionConstants::MOMENTUMGRID_TYPE_PXI;
+
+    DREAM::CoulombLogarithm lnLEE(grid,unknowns,ionHandler,gridtype,cq,DREAM::CollisionQuantity::LNLAMBDATYPE_EE);
+    DREAM::CoulombLogarithm lnLEI(grid,unknowns,ionHandler,gridtype,cq,DREAM::CollisionQuantity::LNLAMBDATYPE_EI);
+    DREAM::SlowingDownFrequency nuS(grid,unknowns,ionHandler,&lnLEE,&lnLEI,gridtype,cq);
+    // DREAM::PitchScatterFrequency *nuD = new DREAM::PitchScatterFrequency(grid,unknowns,ionHandler,lnLEI,lnLEE,gridtype,cq);
```

Comment:
If you do
nuS.RebuildRadialTerms()
here on 133, which would allow you to use GetMeanExcitationEnergy on 139 "as intended" originally (but I don't really mind this either)